### PR TITLE
Add PR shepherd operational intake

### PR DESCRIPTION
**Pull request by KjellKod**
## Summary
Add deterministic PR shepherd plumbing so existing PRs can be targeted, inspected, deduped, replied to, and classified with less agent guesswork.
- Keeps `pr-assistant` responsible for draft PR creation while `pr-shepherd` owns existing PR lifecycle work.
- Adds compact records intake, failed-log summaries, scope annotation, durable reply markers, and operational stop states.

## Changes
- **Skills**:
  - Updates `.skills/pr-shepherd/SKILL.md`, `.agents/skills/pr-shepherd/SKILL.md`, `.claude/skills/pr-shepherd/SKILL.md`, and `.skills/SKILLS.md` so shepherding starts from an existing PR and only marks ready from `clean`.
- **PR Shepherd Helpers**:
  - Adds `scripts/pr_shepherd_checkout.py` for inspection-first PR targeting with explicit `--apply` checkout mutation.
  - Adds `scripts/pr_shepherd_collect_intake.py`, `scripts/pr_shepherd_fetch_failed_logs.py`, `scripts/pr_shepherd_annotate_scope.py`, and `scripts/pr_shepherd_post_reply.py` for compact intake, failed-log records, diff scope, and idempotent replies.
  - Adds `scripts/quest_runtime/pr_shepherd.py` for marker parsing, fingerprints, activity state, summary rendering, and operational-state classification.
- **Review Intelligence**:
  - Extends `normalize-pr-intake` to accept authoritative `records` while preserving metadata and filtering already-addressed shepherd history.
  - Extends `classify-pr-stop` with optional `--pass-facts` / `--operational-output` for `clean`, `progressing`, and `stuck` pass state.
- **Manifest and Docs**:
  - Adds new helper scripts to `.quest-manifest` and documents the pipeline in `scripts/README.md`.
  - Archives the implemented idea and records the Quest journal/celebration artifacts.
- **Tests**:
  - Adds focused unit coverage for checkout guards, marker state, record normalization, failed-log summaries, scope annotation, status-context rollups, activity-state dedupe, and post-reply body handling.

## Validation
- [x] **Unit suite**: Prerequisites: Python test dependencies installed for this repo. Action: run `python3 -m pytest tests/unit` from the repo root. Expected result: all unit tests pass; this branch last passed with `599 passed`. Negative/edge cases: failures in `test_pr_shepherd.py` or `test_pr_review_cycle.py` should be treated as PR-shepherd contract regressions.
- [x] **Manifest check**: Prerequisites: none beyond repo checkout. Action: run `scripts/quest_validate-manifest.sh`. Expected result: both Quest file listing and stale-entry checks pass. Negative/edge cases: any missing helper script means installer propagation would be incomplete.
- [x] **Checkout dry run**: Prerequisites: authenticated `gh` and this PR available. Action: run `python3 scripts/pr_shepherd_checkout.py --json` on this branch. Expected result: JSON with `ok: true`, `action: "none"`, and `target_branch: "ideas/pr-shepherd-operational-intake"`. Negative/edge cases: run with a dirty worktree and no `--apply`; it should still inspect without mutating.
- [x] **Records pipeline walkthrough**: Prerequisites: authenticated `gh` and this PR number. Action: run `python3 scripts/pr_shepherd_collect_intake.py --pr <PR_NUMBER> --output /tmp/pr_intake.json`, then `python3 scripts/quest_review_intelligence.py normalize-pr-intake --input /tmp/pr_intake.json --output /tmp/review_findings.json`. Expected result: both commands complete and `/tmp/review_findings.json` is valid JSON findings. Negative/edge cases: unavailable GitHub data should appear under compact `unavailable` diagnostics, not as silent drops.

Watch for: the new helpers intentionally collect and classify mechanical PR state; semantic review decisions still flow through the existing review-intelligence backlog policy.

## Notes
- This PR was implemented through Quest and the source idea is archived at `ideas/archive/2026-05-14-pr-shepherd-operational-intake.md`.

<pre>
     ▐▛███▜▌
    ▝▜█████▛▘
      ▘▘ ▝▝
Quest/Co-Authored by
Co-Authored-By: Codex <noreply@openai.com>
Co-Authored-By: Claude <noreply@anthropic.com>
in collaboration with KjellKod
</pre>

### CODE REVIEW

File: scripts/pr_shepherd_collect_intake.py [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+        default=[],
+        help="JSON output from pr_shepherd_fetch_failed_logs.py to merge into records",
+    )
+    parser.add_argument("--output")
+    return parser.parse_args()
+
+
+def main() -> int:
+    args = parse_args()
+    failed_log_summaries = [
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/pr_shepherd_collect_intake.py]: Fixed in 63a277b. Failed-log summary files now degrade to unavailable diagnostics on read or parse failures, so intake still returns structured JSON.

File: scripts/pr_shepherd_annotate_scope.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            old_line = int(match.group("old_start"))
+            new_line = int(match.group("new_start"))
+            continue
+        if not current_path or line.startswith("\\"):
+            continue
+        if line.startswith("+") and not line.startswith("+++"):
+            changed.setdefault(current_path, set()).add(new_line)
+            new_line += 1
+        elif line.startswith("-") and not line.startswith("---"):
+            changed.setdefault(current_path, set()).add(old_line)
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/pr_shepherd_annotate_scope.py]: Fixed in 63a277b. Scope annotation no longer treats deleted-side old-line numbers as current changed lines.

File: scripts/quest_runtime/pr_review_cycle.py [hunk 3]
```diff
@@ -111,6 +116,114 @@ def _copy_review_local_index(source: JsonObject, target: JsonObject) -> None:
...
+        if existing is None:
+            merged[key] = copy.deepcopy(finding)
+            continue
+        _copy_optional_metadata(finding, existing)
+        existing["activity_state"] = merge_activity_state(
+            existing.get("activity_state"),
+            finding.get("activity_state"),
+        )
+        labels: list[str] = []
+        for value in (existing.get("source_label"), finding.get("source_label")):
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/quest_runtime/pr_review_cycle.py:167]: Fixed in 63a277b. The duplicate-record merge now preserves all source labels instead of overwriting the earlier label before combining provenance.

File: scripts/pr_shepherd_collect_intake.py [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+                    "unavailable_reason": "parse_failed",
+                    "path": path,
+                    "message": str(exc)[:500],
+                }
+            ],
+        }
+    if not isinstance(payload, dict):
+        return {
+            "records": [],
+            "unavailable": [
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/pr_shepherd_collect_intake.py:303]: Fixed in 42af4f9. Failed-log summary loading now catches invalid UTF-8 and records a decode_failed unavailable diagnostic instead of crashing intake.

File: scripts/pr_shepherd_checkout.py [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+    if not apply:
+        payload["action"] = "would_checkout"
+        return 0, payload
+
+    if not clean:
+        payload.update({"ok": False, "reason": "dirty_worktree", "action": "none"})
+        return 1, payload
+
+    root = _repo_root()
+    if root is not None and ".worktrees" in root.parts:
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/pr_shepherd_checkout.py]: Fixed in 4716a8c. The checkout mutation guard now uses Git worktree metadata from rev-parse instead of matching path names, and the tests cover linked worktrees outside a .worktrees path.

- Reviewed by GPT-5, in collaboration with KjellKod

File: scripts/pr_shepherd_post_reply.py [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+    if args.summary:
+        rows = _load_json(args.summary_rows) or []
+        if not isinstance(rows, list):
+            raise ValueError("--summary-rows must contain a JSON list")
+        body = compact_summary_body(rows) if rows else append_marker(body, SUMMARY_MARKER)
+        comments = _load_json(args.comments_json)
+        if comments is None and not args.dry_run:
+            if not args.pr:
+                raise ValueError("--pr is required when posting a summary")
+            comments, comments_error = _gh_json(["gh", "api", f"repos/{{owner}}/{{repo}}/issues/{args.pr}/comments"])
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/pr_shepherd_post_reply.py]: Fixed in 4716a8c. Summary upsert now scans bounded issue-comment pages before deciding whether to update or create the marker-owned shepherd summary, with a pagination regression test.

- Reviewed by GPT-5, in collaboration with KjellKod

File: scripts/quest_runtime/pr_review_cycle.py [hunk 7]
```diff
@@ -111,6 +116,113 @@ def _copy_review_local_index(source: JsonObject, target: JsonObject) -> None:
...
+        "scope_reason",
+        "author_kind",
+    ):
+        value = source.get(key)
+        if value not in (None, "", [], {}):
+            target[key] = copy.deepcopy(value)
+
+
+def _finding_key(finding: JsonObject) -> tuple[str, str, str, str]:
+    line_value = finding.get("line")
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/quest_runtime/pr_review_cycle.py:135]: Fixed in 4716a8c. The check-run record now carries the check name into the canonical summary before dedupe, with regression coverage for distinct failed checks that otherwise share the same synthetic path and state.

- Reviewed by GPT-5, in collaboration with KjellKod

File: scripts/quest_runtime/pr_shepherd.py [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+    return 0
+
+
+def _bool_fact(pass_facts: JsonObject, key: str) -> bool:
+    return pass_facts.get(key) is True
+
+
+def classify_operational_state(loop_result: JsonObject, pass_facts: JsonObject) -> JsonObject:
+    """Classify a whole PR shepherd pass as clean, progressing, or stuck."""
+
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/quest_runtime/pr_shepherd.py:143]: Fixed in c23d778. The classify-pr-stop CLI now passes its --ci-state into operational classification when pass facts omit ci_state, preserving the command contract for green successful loops.

- Reviewed by GPT-5, in collaboration with KjellKod

File: scripts/pr_shepherd_checkout.py [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+        "action": "none",
+        "target_pr": target_pr,
+        "target_branch": target_branch,
+        "current_branch": current_branch,
+        "worktree_clean": clean,
+        "reason": "",
+        "url": pr.get("url") or "",
+    }
+
+    if current_branch == target_branch:
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/pr_shepherd_checkout.py:113]: Fixed in c23d778. Explicit PR targets with the same local branch name now compare the local HEAD to the PR head OID before no-oping, so stale or wrong local branches do not pass by branch name alone.

- Reviewed by GPT-5, in collaboration with KjellKod

File: scripts/pr_shepherd_post_reply.py [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+            ["gh", "api", endpoint, "--method", "GET", "-F", f"per_page={PER_PAGE}", "-F", f"page={page}"]
+        )
+        if error:
+            return comments, error
+        if not isinstance(payload, list):
+            return comments, "unexpected comments payload"
+        comments.extend(comment for comment in payload if isinstance(comment, dict))
+        if len(payload) < PER_PAGE:
+            return comments, ""
+    return comments, "pagination_truncated"
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/pr_shepherd_post_reply.py]: Fixed in c23d778. The summary-comment scan now checks one page past the configured cap, so an exactly full final page is allowed while a genuinely truncated result still reports truncation.

- Reviewed by GPT-5, in collaboration with KjellKod

File: scripts/quest_runtime/pr_review_cycle.py [hunk 11]
```diff
@@ -111,6 +116,122 @@ def _copy_review_local_index(source: JsonObject, target: JsonObject) -> None:
...
+    }
+    activity = record.get("activity_state")
+    if activity not in (None, "", [], {}):
+        finding["activity_state"] = copy.deepcopy(activity)
+    _copy_optional_metadata(record, finding)
+    _copy_review_local_index(record, finding)
+    return finding
+
+
+def _record_is_actionable(record: JsonObject) -> bool:
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/quest_runtime/pr_review_cycle.py:225]: Fixed in c23d778. Normalization now parses marker-owned shepherd summary fingerprints and applies terminal summary states to matching records before filtering the summary record itself.

- Reviewed by GPT-5, in collaboration with KjellKod

File: scripts/pr_shepherd_checkout.py [hunk 12]
```diff
@@ -0,0 +1,176 @@
...
+        target_oid = str(pr.get("headRefOid") or "")
+        if target and target_oid and _head_oid() != target_oid:
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/pr_shepherd_checkout.py]: Fixed in ae79fd7. When a same-branch head mismatch is applied, the helper now calls gh pr checkout with --force so the stale local branch is actually reset to the PR head.

- Reviewed by GPT-5, in collaboration with KjellKod

File: scripts/pr_shepherd_checkout.py [hunk 13]
```diff
@@ -0,0 +1,176 @@
...
+            return 0, payload
+
+    if current_branch == target_branch:
+        payload["reason"] = "head_mismatch"
+    else:
+        payload["action"] = "would_checkout"
+        if not apply:
+            return 0, payload
+
+    if current_branch == target_branch and not apply:
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/pr_shepherd_checkout.py]: Cleaned up in ae79fd7. The unreachable same-branch dry-run check was removed while tightening the head-mismatch flow.

- Reviewed by GPT-5, in collaboration with KjellKod

File: scripts/pr_shepherd_post_reply.py (hunk 10 above)
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/pr_shepherd_post_reply.py]: Confirmed in ae79fd7 as well. The scanner now fetches one extra page beyond the cap, so exactly-full capped pages no longer report truncation while real overflow still does.

- Reviewed by GPT-5, in collaboration with KjellKod

File: scripts/pr_shepherd_checkout.py (hunk 12 above)
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/pr_shepherd_checkout.py]: Fixed in ae79fd7. The headRefOid comparison now also runs for the current-branch/no-target path, so a stale checkout no longer no-ops just because the branch name matches.

- Reviewed by GPT-5, in collaboration with KjellKod

File: scripts/quest_runtime/pr_review_cycle.py [hunk 14]
```diff
@@ -111,6 +116,155 @@ def _copy_review_local_index(source: JsonObject, target: JsonObject) -> None:
...
+def _apply_summary_addressed_fingerprints(records: list[JsonObject]) -> list[JsonObject]:
+    addressed = _summary_addressed_fingerprints(records)
+    if not addressed:
+        return records
+
+    updated: list[JsonObject] = []
+    for record in records:
+        candidate = copy.deepcopy(record)
+        fingerprint = str(candidate.get("fingerprint") or "")
+        if fingerprint and any(fingerprint.startswith(prefix) for prefix in addressed):
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/quest_runtime/pr_review_cycle.py]: Fixed in ae79fd7. Summary fingerprints now apply only to fingerprint-only records without a thread reply target, so live review-thread state is not overwritten.

- Reviewed by GPT-5, in collaboration with KjellKod

File: scripts/quest_runtime/pr_review_cycle.py [hunk 15]
```diff
@@ -111,6 +116,155 @@ def _copy_review_local_index(source: JsonObject, target: JsonObject) -> None:
...
+
+def _apply_summary_addressed_fingerprints(records: list[JsonObject]) -> list[JsonObject]:
+    addressed = _summary_addressed_fingerprints(records)
+    if not addressed:
+        return records
+
+    updated: list[JsonObject] = []
+    for record in records:
+        candidate = copy.deepcopy(record)
+        fingerprint = str(candidate.get("fingerprint") or "")
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/quest_runtime/pr_review_cycle.py]: Fixed in ae79fd7. Summary fingerprint matching now uses the same stable_fingerprint fallback for records that omit an explicit fingerprint.

- Reviewed by GPT-5, in collaboration with KjellKod

File: scripts/quest_runtime/pr_review_cycle.py [hunk 16]
```diff
@@ -111,6 +116,160 @@ def _copy_review_local_index(source: JsonObject, target: JsonObject) -> None:
...
+def _apply_summary_addressed_fingerprints(records: list[JsonObject]) -> list[JsonObject]:
+    addressed = _summary_addressed_fingerprints(records)
+    if not addressed:
+        return records
+
+    updated: list[JsonObject] = []
+    for record in records:
+        candidate = copy.deepcopy(record)
+        fingerprint = str(candidate.get("fingerprint") or "").strip() or stable_fingerprint(candidate)
+        has_thread_state = candidate.get("reply_target") not in (None, "", [], {})
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/quest_runtime/pr_review_cycle.py]: Fixed in 3177bdb. Summary fingerprint resolution now excludes only real review-comment thread targets; top-level issue comments can still be resolved through the marker-owned summary path.

- Reviewed by GPT-5, in collaboration with KjellKod

File: scripts/pr_shepherd_collect_intake.py [hunk 17]
```diff
@@ -0,0 +1,354 @@
...
+    if not isinstance(checks, list):
+        return records
+    for check in checks:
+        if not isinstance(check, dict):
+            continue
+        conclusion = str(check.get("conclusion") or "").lower()
+        status = str(check.get("status") or "").lower()
+        context_state = str(check.get("state") or "").lower()
+        state = conclusion or status or context_state or "unknown"
+        if state == "success":
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/pr_shepherd_collect_intake.py]: Fixed in 3177bdb. Check-run intake now emits records only for terminal failure states; pending, queued, in-progress, skipped, and success states remain operational wait/status data instead of actionable findings.

- Reviewed by GPT-5, in collaboration with KjellKod

File: scripts/pr_shepherd_checkout.py [hunk 18]
```diff
@@ -0,0 +1,174 @@
...
+        return 1, payload
+
+    if _is_linked_worktree():
+        payload.update({"ok": False, "reason": "worktree_mismatch", "action": "none"})
+        return 1, payload
+
+    checkout_target = target or str(target_pr)
+    checkout_args = ["gh", "pr", "checkout", checkout_target]
+    if payload.get("reason") == "head_mismatch":
+        checkout_args.append("--force")
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/pr_shepherd_checkout.py]: Fixed in 3177bdb. Same-branch head mismatch no longer force-resets; apply mode now refuses the mutation with a head_mismatch blocker so local-only commits are not overwritten.

- Reviewed by GPT-5, in collaboration with KjellKod

File: scripts/pr_shepherd_checkout.py (hunk 13 above)
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/pr_shepherd_checkout.py]: Still covered after 3177bdb. The unreachable same-branch dry-run branch remains removed.

- Reviewed by GPT-5, in collaboration with KjellKod

File: scripts/quest_runtime/pr_review_cycle.py (hunk 16 above)
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/quest_runtime/pr_review_cycle.py]: Fixed in 3177bdb. Summary resolution now distinguishes review_comment thread targets from issue_comment summary targets, so addressed top-level PR comments can resolve through summary fingerprints.

- Reviewed by GPT-5, in collaboration with KjellKod

File: scripts/pr_shepherd_post_reply.py (hunk 10 above)
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/pr_shepherd_post_reply.py]: Still covered after 3177bdb. The exactly-full page-cap case remains covered by the one-extra-page scan and regression test.

- Reviewed by GPT-5, in collaboration with KjellKod

File: scripts/pr_shepherd_checkout.py (hunk 12 above)
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/pr_shepherd_checkout.py]: Still covered after 3177bdb. The no-target path now checks headRefOid and refuses stale same-branch state instead of no-oping.

- Reviewed by GPT-5, in collaboration with KjellKod

[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/pr_shepherd_checkout.py]: Updated in 3177bdb. Rather than force-resetting a stale same-branch checkout, apply mode now refuses with head_mismatch so local-only commits are preserved.

- Reviewed by GPT-5, in collaboration with KjellKod

File: scripts/quest_runtime/pr_review_cycle.py (hunk 15 above)
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/quest_runtime/pr_review_cycle.py]: Still covered after 3177bdb. Records without explicit fingerprints still use the stable_fingerprint fallback for summary matching.

- Reviewed by GPT-5, in collaboration with KjellKod

File: scripts/quest_runtime/pr_review_cycle.py (hunk 14 above)
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/quest_runtime/pr_review_cycle.py]: Still covered after 3177bdb. Summary fingerprints do not override live review-comment thread state; only non-thread summary targets can be resolved this way.

- Reviewed by GPT-5, in collaboration with KjellKod

File: scripts/pr_shepherd_collect_intake.py [hunk 19]
```diff
@@ -0,0 +1,354 @@
...
+        }
+        record["fingerprint"] = stable_fingerprint(record)
+        records.append(record)
+    return records
+
+
+def _issue_comment_record(comment: dict[str, Any]) -> dict[str, Any]:
+    author, author_kind = _author(comment)
+    body = str(comment.get("body") or "")
+    source_kind = "shepherd_summary" if has_marker(body, SUMMARY_MARKER) else "issue_comment"
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/pr_shepherd_collect_intake.py]: Fixed in 8762b08. The collector now treats hidden summary markers as shepherd-owned only when the issue comment is not human-authored; human-authored marker comments remain ordinary issue_comment records. Added regression coverage for both human marker spoofing and bot-authored shepherd summaries.

Validation: python3 -m pytest tests/unit/test_pr_shepherd.py; python3 -m pytest tests/unit; bash scripts/quest_validate-quest-config.sh.

- Reviewed by GPT-5, in collaboration with KjellKod

File: scripts/quest_runtime/pr_shepherd.py [hunk 20]
```diff
@@ -0,0 +1,216 @@
...
+    """Classify marker recency for one thread/comment/fingerprint activity list."""
+
+    ordered = sorted(
+        activities,
+        key=lambda item: str(item.get("created_at") or item.get("updated_at") or ""),
+    )
+    last_marker_index: int | None = None
+    for index, activity in enumerate(ordered):
+        body = str(activity.get("body") or "")
+        if has_marker(body, ADDRESSED_MARKER) or has_marker(body, SUMMARY_MARKER):
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/quest_runtime/pr_shepherd.py]: Fixed in 70a6e3b. Marker recognition is now gated by collector-provided marker trust: bot markers are trusted, the current shepherding GitHub user is trusted, and untrusted human-authored marker strings are ignored by activity_state.

Validation:
- python3 -m pytest tests/unit/test_pr_shepherd.py
- python3 -m pytest tests/unit
- bash scripts/quest_validate-quest-config.sh

- Reviewed by GPT-5, in collaboration with KjellKod

File: scripts/quest_runtime/pr_shepherd.py [hunk 21]
```diff
@@ -0,0 +1,216 @@
...
+        if _bool_fact(pass_facts, key):
+            hard_blockers.append(key)
+    if _bool_fact(pass_facts, "loop_cap_enforced"):
+        hard_blockers.append("loop_cap_enforced")
+    if human_decisions:
+        hard_blockers.append("human_decision_required")
+    if ci_state in {"failing", "unknown"}:
+        hard_blockers.append(f"ci_{ci_state}")
+    if active or uncertain:
+        hard_blockers.append("feedback_remaining")
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/quest_runtime/pr_shepherd.py:167]: Fixed in 70a6e3b. Remaining feedback is now a stuck blocker only when the pass made no safe progress; if the pass posted replies or pushed commits, the operational state can report progressing while waiting for the next CI/review cycle.

Validation:
- python3 -m pytest tests/unit/test_pr_shepherd.py
- python3 -m pytest tests/unit
- bash scripts/quest_validate-quest-config.sh

- Reviewed by GPT-5, in collaboration with KjellKod

File: scripts/quest_runtime/pr_shepherd.py [hunk 22]
```diff
@@ -0,0 +1,216 @@
...
+    encoded = json.dumps(relevant, sort_keys=True, separators=(",", ":"))
+    return hashlib.sha256(encoded.encode("utf-8")).hexdigest()
+
+
+def activity_state(activities: list[JsonObject]) -> str:
+    """Classify marker recency for one thread/comment/fingerprint activity list."""
+
+    ordered = sorted(
+        activities,
+        key=lambda item: str(item.get("created_at") or item.get("updated_at") or ""),
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/quest_runtime/pr_shepherd.py]: Fixed in 70a6e3b. activity_state now orders each activity by the latest of created_at and updated_at, so an edited earlier human comment can reactivate feedback after a shepherd marker.

Validation:
- python3 -m pytest tests/unit/test_pr_shepherd.py
- python3 -m pytest tests/unit
- bash scripts/quest_validate-quest-config.sh

- Reviewed by GPT-5, in collaboration with KjellKod

File: scripts/pr_shepherd_collect_intake.py [hunk 23]
```diff
@@ -0,0 +1,363 @@
...
+    record = {
+        "source_kind": source_kind,
+        "source_label": "github-pr-comment",
+        "activity_state": activity_state([_activity(comment)]),
+        "author": author,
+        "author_kind": author_kind,
+        "path": "pr/comment",
+        "line": None,
+        "body_excerpt": _excerpt(body),
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/pr_shepherd_collect_intake.py:181]: Fixed in 70a6e3b. Shepherd summary records now preserve the full comment body while still keeping the bounded body_excerpt for compact display, so durable fingerprint rows are not lost to excerpt truncation.

Validation:
- python3 -m pytest tests/unit/test_pr_shepherd.py
- python3 -m pytest tests/unit
- bash scripts/quest_validate-quest-config.sh

- Reviewed by GPT-5, in collaboration with KjellKod

File: scripts/pr_shepherd_checkout.py [hunk 24]
```diff
@@ -0,0 +1,179 @@
...
+        payload.update({"ok": False, "reason": "dirty_worktree", "action": "none"})
+        return 1, payload
+
+    if _is_linked_worktree():
+        payload.update({"ok": False, "reason": "worktree_mismatch", "action": "none"})
+        return 1, payload
+
+    checkout_target = target or str(target_pr)
+    result = _run(["gh", "pr", "checkout", checkout_target])
+    if result.returncode != 0:
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/pr_shepherd_checkout.py:145]: Fixed in 70a6e3b. After an applied gh pr checkout, the helper now verifies local HEAD against the PR head OID and fails with head_mismatch if the checkout landed on stale local state.

Validation:
- python3 -m pytest tests/unit/test_pr_shepherd.py
- python3 -m pytest tests/unit
- bash scripts/quest_validate-quest-config.sh

- Reviewed by GPT-5, in collaboration with KjellKod

File: scripts/pr_shepherd_collect_intake.py [hunk 25]
```diff
@@ -0,0 +1,363 @@
...
+PER_PAGE = 100
+CHECK_FAILURE_STATES = {
+    "action_required",
+    "cancelled",
+    "error",
+    "failure",
+    "failed",
+    "startup_failure",
+    "timed_out",
+}
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/pr_shepherd_collect_intake.py:30]: Fixed in 70a6e3b. The intake check failure allowlist now includes GitHub stale conclusions, with coverage proving stale check runs produce check_run records instead of disappearing.

Validation:
- python3 -m pytest tests/unit/test_pr_shepherd.py
- python3 -m pytest tests/unit
- bash scripts/quest_validate-quest-config.sh

- Reviewed by GPT-5, in collaboration with KjellKod

File: scripts/pr_shepherd_post_reply.py [hunk 26]
```diff
@@ -0,0 +1,180 @@
...
+            return comments, "pagination_truncated" if payload else ""
+        comments.extend(comment for comment in payload if isinstance(comment, dict))
+        if len(payload) < PER_PAGE:
+            return comments, ""
+    return comments, ""
+
+
+def _find_summary_comment(comments: list[dict[str, Any]]) -> dict[str, Any] | None:
+    for comment in comments:
+        if has_marker(str(comment.get("body") or ""), SUMMARY_MARKER):
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/pr_shepherd_post_reply.py]: Fixed in c38c36c. Summary upsert now gates marker ownership by trusted author: bot-authored summaries and the current shepherd GitHub user are trusted, while unrelated human comments containing the marker are ignored.

Validation:
- python3 -m pytest tests/unit/test_pr_shepherd.py
- python3 -m pytest tests/unit
- bash scripts/quest_validate-quest-config.sh

- Reviewed by GPT-5, in collaboration with KjellKod

File: scripts/pr_shepherd_collect_intake.py [hunk 27]
```diff
@@ -0,0 +1,398 @@
...
+    return record
+
+
+def _review_body_record(review: dict[str, Any], *, trusted_marker_author: str = "") -> dict[str, Any] | None:
+    body = str(review.get("body") or "").strip()
+    if not body:
+        return None
+    author, author_kind = _author(review)
+    record = {
+        "source_kind": "review_body_item",
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/pr_shepherd_collect_intake.py:200]: Fixed in c38c36c. Intake now skips approved and dismissed review bodies, so approval text like LGTM does not become active PR feedback.

Validation:
- python3 -m pytest tests/unit/test_pr_shepherd.py
- python3 -m pytest tests/unit
- bash scripts/quest_validate-quest-config.sh

- Reviewed by GPT-5, in collaboration with KjellKod

File: scripts/pr_shepherd_collect_intake.py [hunk 28]
```diff
@@ -0,0 +1,401 @@
...
+    user = comment.get("user")
+    if not isinstance(user, dict):
+        return "", "unknown"
+    login = str(user.get("login") or "")
+    user_type = str(user.get("type") or "").lower()
+    return login, "bot" if user_type == "bot" or login.endswith("[bot]") else "human"
+
+
+def _trusted_marker_author(author: str, author_kind: str, trusted_marker_author: str) -> bool:
+    return author_kind != "human" or bool(trusted_marker_author and author == trusted_marker_author)
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/pr_shepherd_collect_intake.py]: Fixed in 487fcf5. `_trusted_marker_author()` now only trusts bot authors or the configured human marker author, and `unknown` authors remain ordinary comments. Added regression coverage in `test_collect_intake_ignores_unknown_authored_summary_markers()` and validated with `python3 -m pytest tests/unit/test_pr_shepherd.py`, `python3 -m pytest tests/unit`, and `bash scripts/quest_validate-quest-config.sh`.
